tests/impl.cpp

```diff
@@ -6235,7 +6235,18 @@ result_t test_mm_shufflelo_epi16_function(const SSE2NEONTestImpl &impl,
 
 result_t test_mm_cvtps_pi16(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    const float *_a = impl.mTestFloatPointer1;
+    float _b[4];
+    int16_t trun[4];
+    
+    for (int j = 0; j < 4; j++) {
+        _b[j] = fabsf(_a[j]) > 32767.0f ? _a[j] / 10.0f : _a[j];
```

Comment:
What is `_a[j] / 10.0f` for?

Reply:
@JishinMaster let's discuss under this conversation.

First, I would like to know the behaviour of the SSE when it is beyond the range of an `int16_t`.
That is, please give more description about `Neither the NEON version nor the original SSE behave well beyond the range of an int16_t.` .

I prefer the test should take the situation of `Note: this intrinsic will generate 0x7FFF, rather than 0x8000, for input values between 0x7FFF and 0x7FFFFFFF` into consideration.

Since divide with the value `10.0` is not intuitive, at least add the comment to describe the reason for doing this.

Reply:
If I remove the /10.0f patch, here is what I have on my Atom CPU :
In -68298.343750 Out -32768 Ref -2762
In 44244.390625 Out 32767 Ref -21292
In -61859.132812 Out -32768 Ref 3677
In 8441.164062 Out 8441 Ref 8441
In -68298.343750 Out -32768 Ref -2762
In 44244.390625 Out 32767 Ref -21292
In -61859.132812 Out -32768 Ref 3677
In 8441.164062 Out 8441 Ref 8441

The "Out" is the value after _mm_cvtps_pi16, and "Ref" is using (int16_t)bankersRounding.
For input values above the range of int16_t ( > 32767.0f or < -32768.0f), _mm_cvtps_pi16 saturates at either 32767 or 32768 (depending on the sign).
The aarch64 sse2neon port (using Qemu) gives the same results as the Atom CPU as long as the input value stay in the range of int16_t. Outside this range values may differ.

I will add a comment so it is clearer.